File: php/testcode/codeinj_033.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_codeinj_ob_callback_eval
function codeinj033(BenchmarkRequest $req): BenchmarkResponse {
    $extra = $req->param('extra');
    ob_start(function ($buffer) use ($extra) {
        return eval($buffer . $extra); // vuln-code-snippet vuln-line php_codeinj_ob_callback_eval
    });
    echo 'return ';
    $output = ob_get_clean();
    return BenchmarkResponse::ok((string) $output);
}
// vuln-code-snippet end php_codeinj_ob_callback_eval

File: ruby/testcode/codeinj_033.rb
require_relative 'shared'

# vuln-code-snippet start ruby_codeinj_template_eval
def dispatch_method(req)
  method_name = req.param('method_name')
  ctx = binding
  result = ctx.eval("self.#{method_name}") # vuln-code-snippet vuln-line ruby_codeinj_template_eval
  BenchmarkResponse.json({ result: result.to_s })
end
# vuln-code-snippet end ruby_codeinj_template_eval
